3-networks/modules/restricted_shared_vpc/main.tf

```diff
@@ -0,0 +1,210 @@
+ *****************************************/
+
+locals {
+  vpc_type                   = "shared"
+  vpc_label                  = (var.vpc_label != "" ? var.vpc_label : "restricted")
+  vpc_name                   = "${var.environment_code}-${local.vpc_type}-${local.vpc_label}"
+  network_name               = "vpc-${local.vpc_name}"
+  restricted_googleapis_cidr = "199.36.153.4/30"
+  subnets = [
+    for subnet in var.subnets : {
+      subnet_name           = "sb-${local.vpc_name}-${subnet.subnet_region}"
+      subnet_ip             = subnet.subnet_ip
+      subnet_region         = subnet.subnet_region
+      subnet_private_access = subnet.subnet_private_access
+      subnet_flow_logs      = subnet.subnet_flow_logs
+      description           = subnet.description
+    }
+  ]
+
+  temp_secondary_ranges = [
+    for subnet in var.subnets : {
+      sub_name = "sb-${local.vpc_name}-${subnet.subnet_region}"
+      ranges = [
+        for range in subnet.secondary_ranges : {
+          range_name    = "rn-${local.vpc_name}-${subnet.subnet_region}-${range.range_label}"
+          ip_cidr_range = range.ip_cidr_range
+        }
+      ]
+    }
+  ]
+
+  secondary_ranges = {
+    for ranges in local.temp_secondary_ranges :
+    ranges.sub_name => ranges.ranges
+  }
+
+  bgp_asn = {
+    for subnet in var.subnets :
+    "sb-${local.vpc_name}-${subnet.subnet_region}" => subnet.bgp_asn
+  }
+}
+
+module "main" {
+  source                                 = "terraform-google-modules/network/google"
+  version                                = "~> 2.0"
+  project_id                             = var.project_id
+  network_name                           = local.network_name
+  shared_vpc_host                        = "true"
+  delete_default_internet_gateway_routes = "true"
+
+  subnets          = local.subnets
+  secondary_ranges = local.secondary_ranges
+
+  routes = [
+    {
+      name              = "rt-${local.vpc_name}-1000-all-default-restricted-api"
+      description       = "Route through IGW to allow restricted google api access."
+      destination_range = local.restricted_googleapis_cidr
+      next_hop_internet = "true"
+      priority          = "1000"
+    }
+  ]
+}
+
+/***************************************************************
+  Configure Service Networking for Cloud SQL & future services.
+ **************************************************************/
+
+resource "google_compute_global_address" "private_service_access_address" {
+  provider = google-beta
+
+  name          = "ga-${local.vpc_name}-vpc-peering-internal"
+  project       = var.project_id
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  address       = element(split("/", var.private_service_cidr), 0)
+  prefix_length = element(split("/", var.private_service_cidr), 1)
+  network       = module.main.network_self_link
+}
+
+resource "google_service_networking_connection" "private_vpc_connection" {
+  provider                = google-beta
```

Comment:
Are we using any beta features in these resources where we are using the beta provider?

Reply:
This is likely a relic of the past, from when this resource was in beta

Reply:
`provider = google-beta` removed from resources.